cumulus_library/base_table_builder.py

```diff
@@ -0,0 +1,49 @@
+""" abstract base for python-based study executors """
+from abc import ABC, abstractmethod
+from typing import final
+
+from cumulus_library.helper import get_progress_bar, query_console_output
+
+
+class BaseTableBuilder(ABC):
+    """Generic base class for python table builders.
+
+    To use a table builder, extend this class exactly once in a new module.
+    See ./studies/vocab or ./studies/core for example usage.
+    """
+
+    def __init__(self):
+        self.queries = []
+        self.display_text = "Building custom tables..."
```

Comment:
nit shower thought: you could make this a class variable, so that subclasses don't have to define `__init__` at all, saving some boilerplate. They could still define it per-instance if they really wanted, but I bet the 99% case is a static string.

Reply:
Naw, my comment was less about best practices and more "I think this will be easier for users" but just a nit